features/flows/headers/vxlan/test_vxlan_header_with_inner_ipv6_traffic.py

```diff
@@ -0,0 +1,189 @@
+import logging as log
+import pytest
+from helpers.otg import otg
+
+
+@pytest.mark.all
+@pytest.mark.feature
+@pytest.mark.b2b
+@pytest.mark.dp_feature
+def test_vxlan_header_with_inner_ipv6_traffic():
+    test_const = {
+        "pktRate": 50,
+        "pktCount": 100,
+        "pktSize": 256,
+        "txMac": "00:00:01:01:01:01",
+        "rxMac": "00:00:01:01:01:02",
+        "innerTxMac": "00:00:01:01:01:03",
+        "innerRxMac": "00:00:01:01:01:04",
+        "txIp": "1.1.1.1",
+        "rxIp": "1.1.1.2",
+        "txUdpPortValue": 4789,
+        "rxUdpPortValue": 4789,
+        "vxLanVniValues": [1000, 1001, 1002, 1003, 1004],
+        "txTcpPortValue": 80,
+        "rxTcpPortValue": 80,
+    }
+    api = otg.OtgApi()
+    c = vxlan_header_config(api, test_const)
+
+    api.set_config(c)
+
+    api.start_capture()
+    api.start_transmit()
+
+    api.wait_for(
+        fn=lambda: metrics_ok(api, test_const), fn_name="wait_for_flow_metrics"
+    )
+
+    api.stop_capture()
+
+    capture_ok(api, c, test_const)
+
+
+def vxlan_header_config(api, tc):
+    c = api.api.config()
+    p1 = c.ports.add(name="p1", location=api.test_config.otg_ports[0])
+    p2 = c.ports.add(name="p2", location=api.test_config.otg_ports[1])
+
+    ly = c.layer1.add(name="ly", port_names=[p1.name, p2.name])
+    ly.speed = api.test_config.otg_speed
+
+    if api.test_config.otg_capture_check:
+        ca = c.captures.add(name="ca", port_names=[p1.name, p2.name])
+        ca.format = ca.PCAP
+
+    f1 = c.flows.add(name="f1")
+    f1.tx_rx.port.tx_name = p1.name
+    f1.tx_rx.port.rx_name = p2.name
+    f1.duration.fixed_packets.packets = tc["pktCount"]
+    f1.rate.pps = tc["pktRate"]
+    f1.size.fixed = tc["pktSize"]
+    f1.metrics.enable = True
+
+    eth1, ip, udp, vxlan, eth2, ip6, tcp = (
+        f1.packet.ethernet().ipv4().udp().vxlan().ethernet().ipv6().tcp()
+    )
+
+    eth1.src.value = tc["txMac"]
+    eth1.dst.value = tc["rxMac"]
+
+    eth2.src.value = tc["innerTxMac"]
+    eth2.dst.value = tc["innerRxMac"]
+
+    ip.src.value = tc["txIp"]
```

Comment:
Please add config and validation for tx/rx IPv6 address

Reply:
added config as well as validation for both ip versions